app/models/service_template_ansible_playbook.rb

```diff
@@ -8,29 +8,31 @@ def self.default_retirement_entry_point
   end
 
   # create ServiceTemplate and supporting ServiceResources and ResourceActions
-  # options
+  # item_def
   #   :name
   #   :description
-  #   :service_template_catalog
+  #   :service_template_catalog_id
   #   :config_info
   #     :provision
-  #       :service_dialog_id (or)
+  #       :dialog_id (or)
   #       :new_dialog_name
-  #       :variables
+  #       :extra_vars
   #       :hosts
-  #       :credentials
+  #       :credential_id
+  #       :cloud_credential_id
+  #       :network_credential_id
   #       :playbook_id
   #     :retirement (same as provision)
   #     :reconfigure (same as provision)
   #
-  def self.create_catalog_item(options, _auth_user)
-    options      = options.merge(:service_type => 'atomic', :prov_type => 'generic_ansible_playbook')
-    service_name = options[:name]
-    description  = options[:description]
-    config_info  = validate_config_info(options)
+  def self.create_catalog_item(item_def, _auth_user)
+    item_def     = item_def.merge(:service_type => 'atomic', :prov_type => 'generic_ansible_playbook')
+    service_name = item_def[:name]
+    description  = item_def[:description]
+    config_info  = validate_config_info(item_def)
 
     transaction do
-      create(options.except(:config_info)).tap do |service_template|
+      create(item_def.except(:config_info).merge(:options => {:catalog_item => item_def})).tap do |service_template|
```

Comment:
@jntullo I was thinking about the same 😄 We need the base `ServiceTemplate` to have the common code to pull attributes from the template itself into a formatted hash, and merge the ones read back from options for `config_info`. Since this is a wip PR, I will wait for you to complete the base model. Thanks!